ddtrace/tracer/log_test.go

```diff
@@ -259,3 +259,52 @@ func setup(t *testing.T, customProp Propagator) string {
 	require.Len(t, tp.Logs(), 2)
 	return tp.Logs()[1]
 }
+
+func TestAgentURL(t *testing.T) {
+	assert := assert.New(t)
+	tp := new(log.RecordLogger)
+	tracer := newTracer(WithLogger(tp), WithUDS("var/run/datadog/apm.socket"))
+	defer tracer.Stop()
+	tp.Reset()
+	tp.Ignore("appsec: ", telemetry.LogPrefix)
+	logStartup(tracer)
+	assert.Regexp(`"agent_url":"unix://var/run/datadog/apm.socket"`, tp.Logs()[1])
+}
+
+func TestAgentURLFromEnv(t *testing.T) {
+	assert := assert.New(t)
+	t.Setenv("DD_TRACE_AGENT_URL", "unix://var/run/datadog/apm.socket")
+	tp := new(log.RecordLogger)
+	tracer := newTracer(WithLogger(tp))
+	defer tracer.Stop()
+	tp.Reset()
+	tp.Ignore("appsec: ", telemetry.LogPrefix)
+	logStartup(tracer)
+	assert.Regexp(`"agent_url":"unix://var/run/datadog/apm.socket"`, tp.Logs()[1])
+}
+
+func TestInvalidAgentURL(t *testing.T) {
+	assert := assert.New(t)
+	// invalid socket URL
+	t.Setenv("DD_TRACE_AGENT_URL", "var/run/datadog/apm.socket")
+	tp := new(log.RecordLogger)
+	tracer := newTracer(WithLogger(tp))
+	defer tracer.Stop()
+	tp.Reset()
+	tp.Ignore("appsec: ", telemetry.LogPrefix)
+	logStartup(tracer)
+	// assert that it is the default URL
+	assert.Regexp(`"agent_url":"http://localhost:8126/v0.4/traces"`, tp.Logs()[0])
```

Comment:
Any progress on why we have to access the first index instead of the second here, as opposed to the previous couple tests?